### DISCUSSION TITLE
[Porous Flow] Unstructured Grid Heterogeneous Properties

### DISCUSSION THREAD

**Discussion by jrpatterson79** [idaholab/moose#24968] [Q&A General] [answered]
Hello MOOSE friends,

I would like to apply heterogeneous flow properties to an unstructured grid in a PorousFlow simulation. I've read through the [tutorial developed](https://mooseframework.inl.gov/modules/porous_flow/heterogeneous_models.html) by the PorousFlow collaborators and a [discussion thread](https://github.com/idaholab/moose/discussions/18542) from a couple of years ago and my sense is that the current tools (i.e. PiecewiseMultilinear) available in MOOSE are developed for structured grids. 

My first question is whether this is still true? Is there a Function I'm overlooking that now applies properties to unstructured grids similar to PiecewiseMultilinear and PiecewiseMultiConstant?

A follow-up question. I have a mesh in .vtu format that contains the property of interest defined at cell centroids. If I import that mesh using FileMeshGenerator, is it possible to extract that to an auxiliary variable that I can use within my input file?

Thank y'all in advance for your thoughts

**Comment by 1runer** [comment, marked as answer]
At least for the latter question if the mesh is the same:
You could save the mesh as Exodus .e with Paraview and use a [SolutionUserobject](https://mooseframework.inl.gov/source/userobjects/SolutionUserObject.html) to load the data and transfer it into an AuxVariable and link the AuxVariable to the Materials like Porosity or Permeability. 

With that solution you don't need any interpolation, but the mesh should stay the same.

**Reply by cpgr** [reply]
Yeah, if you can turn it into an exodus file then that is probably the best thing to do. You won't even need a SolutionUserObject, you can initialise elemental AuxVariables directly from the exodus file using 'initial_from_file_var = '. The example you linked does exactly that.

**Reply by jrpatterson79** [reply]
Thank you @1runer and @cpgr 

I see where I overlooked the 'initial_from_file_var' in the linked tutorial
